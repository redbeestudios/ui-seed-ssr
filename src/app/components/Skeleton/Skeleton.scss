@import "./../../assets/css/variables";


.skeleton__content {
    box-shadow: 0 4px 10px 0 rgba(33, 33, 33, 0.15);
    border-radius: 4px;
    position: relative;
    overflow: hidden;
    width: 100%;
    height: 25px;
    background: white;
    margin: 0 0 10px 0;
}

.skeleton__content::before {
    content: '';
    display: block;
    position: absolute;
    left: -150px;
    top: 0;
    height: 100%;
    width: 150px;
    background: linear-gradient(to right, transparent 0%, #E8E8E8 50%, transparent 100%);
    animation: load 1s infinite cubic-bezier(0.4, 0.0, 0.2, 1);
}

@keyframes load {
    from {
        left: -150px;
    }
    to   {
        left: 100%;
    }
}
