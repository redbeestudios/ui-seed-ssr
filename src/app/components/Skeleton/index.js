import React from "react";
import PropTypes from "prop-types";

import "./Skeleton.scss";

const Skeleton = ({ multiply }) => {
  return (
    <div className="skeleton">
      {Array(multiply)
        .fill()
        .map((it) => (
          <div key={`skeleton_${it}`} className="skeleton__content" />
        ))}
    </div>
  );
};

Skeleton.defaultProps = {
  multiply: 1,
};

Skeleton.propTypes = {
  multiply: PropTypes.number,
};

export default Skeleton;
