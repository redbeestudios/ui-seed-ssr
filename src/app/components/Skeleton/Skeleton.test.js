import React from "react";
import { shallow } from "enzyme";
import Skeleton from "./index";

describe("Skeleton", () => {
  it("should render correctly", () => {
    const component = shallow(<Skeleton />);
    expect(component).toMatchSnapshot();
    expect(component.find(".skeleton__content")).toHaveLength(1);
  });

  it("should render a number of times", () => {
    const quantity = 4;
    const component = shallow(<Skeleton multiply={quantity} />);
    expect(component).toMatchSnapshot();
    expect(component.find(".skeleton__content")).toHaveLength(quantity);
  });
});
