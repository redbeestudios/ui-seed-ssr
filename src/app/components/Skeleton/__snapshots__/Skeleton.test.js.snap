// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skeleton should render a number of times 1`] = `
<div
  className="skeleton"
>
  <div
    className="skeleton__content"
    key="0"
  />
  <div
    className="skeleton__content"
    key="1"
  />
  <div
    className="skeleton__content"
    key="2"
  />
  <div
    className="skeleton__content"
    key="3"
  />
</div>
`;

exports[`Skeleton should render correctly 1`] = `
<div
  className="skeleton"
>
  <div
    className="skeleton__content"
    key="0"
  />
</div>
`;
